src/analysis/alt_series.lean

```diff
@@ -0,0 +1,192 @@
+
+/-!
+# Convergence of alternating series
+
+An alternating series is an infinite series of the form
+$\sum_{k=0}^{\infty} (-1)^k a_k$ or $\sum_{k=0}^{\infty} (-1)^{k+1} a_k$,
+where $a_n : \real$ and $a_n \ge 0$.
+
+The **alternating series test** states that an alternating series is convergent if
+$\lim_{n\to\infty} a_n = 0$ and $a_{n+1} \leq a_n$. We prove this via the Cauchy
+convergence test by showing that its partial sums, $s_k = \sum_0^k (-1)^k a_k$,
+form a Cauchy sequence.
+
+Even though they are Cauchy, alternating series do not fulfill the current
+definition of `summable` in mathlib, which requires that a series be
+*absolutely* convergent.  Alternating series are only *conditionally*
+convergent.
+
+## Main definitions
+
+* `alternating_partial_sum` : An alternating series whose first term is positive.
+* `alternating_partial_sum'`: An alternating series whose first term is negative.
+
+## Main results
+
+* `alternating_partial_sum_is_cauchy_seq`: The partial sums of an alternating series
+  form a Cauchy sequence.
+* `alternating_partial_sum_is_cauchy_seq'`: Same as above, but for series whose first
+  term is negative.
+
+## References
+
+* https://en.wikipedia.org/wiki/Alternating_series
+
+## Tags
+
+series, alternating series, Cauchy convergence test
+-/
+
+open nat filter finset
+open_locale big_operators topological_space
+
+variable a : ℕ → ℝ
+
+/-- The partial sum of an alternating series whose first term is positive -/
+def alternating_partial_sum (k : ℕ) := ∑ n in range k, (-1)^n * a n
+
+/-- The partial sum of an alternating series whose first term is negative -/
+def alternating_partial_sum' (k : ℕ) := ∑ n in range k, (-1)^n * -(a n)
+
+lemma alternating_partial_sum_nonneg
+  (ha_anti : antitone a)
+  (ha_nonneg : ∀ n, 0 ≤ a n)
+  (k : ℕ) :
+  0 ≤ alternating_partial_sum a k :=
+begin
+  unfold alternating_partial_sum,
+  induction k using nat.case_strong_induction_on with k hk,
+  { simp only [sum_empty, range_zero] },
+  { obtain heven | hodd := even_or_odd k,
+    { simp only [sum_range_succ, neg_one_pow_of_even heven, one_mul, ge_iff_le],
+      exact add_nonneg (hk k rfl.le) (ha_nonneg k), },
+    { cases k with k,
+      { exfalso, rw odd_iff_not_even at hodd, exact hodd even_zero, },
+      { have : even k := by simpa with parity_simps using hodd,
+        simp only [sum_range_succ, neg_one_pow_of_even this, neg_one_pow_of_odd hodd],
+        rw [one_mul, neg_one_mul, add_assoc],
+        apply add_nonneg (hk k (lt_succ_self k).le),
+        rw [le_add_neg_iff_add_le, zero_add],
+        exact ha_anti (lt_add_one k).le, } } },
+end
+
+lemma anti_suffix_is_anti {m : ℕ}
+  (ha_anti : antitone a) :
+  antitone (λ (n : ℕ), a (m + n)) :=
+λ x y hx, ha_anti (add_le_add_left hx m)
+
+lemma alternating_partial_sum_diff_nonneg
+  {m n : ℕ}
+  (ha_anti : antitone a)
+  (ha_nonneg : ∀ n, 0 ≤ a n)
+  (hmn : m ≤ n)
+  (hm : even m) :
+  0 ≤ alternating_partial_sum a n - alternating_partial_sum a m :=
```

Comment:
Perhaps this would be better stated as `alternating_partial_sum a m ≤ alternating_partial_sum a n`? You can use `sub_nonneg` to get it into this form. Name suggestion: `alternating_partial_sum_even_mono`